### PR TITLE
[CORE-354] - Add tracking for market availability in order to alert in datadog

### PR DESCRIPTION
**Pull request by clemire**
### Changelist
Track availability of market symbols in each exchange for alerting purposes.

### Test Plan
Existing tests, this is a telemetry change.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go [hunk 1]
```diff
@@ -183,6 +184,27 @@ func (pf *PriceFetcher) RunTaskLoop(requestHandler daemontypes.RequestHandler) {
 	}
 }
 
+// logMarketAvailability emits telemetry that tracks whether a market was available when queried on an exchange.
+// Success is tracked by (market, exchange) so that we can track the availability of each market on each exchange.
+func logMarketAvailability(exchangeId types.ExchangeId, id types.MarketId, available bool) {
```
Comment (teddyding) [inline protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go:189]: Nit: `emitMarketAvailabilityMetrics` or something to better reflect the intention of this function?

File: protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go [hunk 2]
```diff
@@ -239,9 +261,21 @@ func (pf *PriceFetcher) runSubTask(
...
+			logMarketAvailability(exchangeId, marketId, false)
+		}
+
 		return
 	}
 
+	// Track which markets were available when queried, and which were not, for telemetry.
+	availableMarkets := make(map[types.MarketId]bool, len(marketIds))
+	for _, marketId := range marketIds {
+		availableMarkets[marketId] = false
```
Comment (teddyding) [inline protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go:276]: Where is the value of `availableMarkets` ever used?

File: protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go [hunk 3]
```diff
@@ -269,8 +303,16 @@ func (pf *PriceFetcher) runSubTask(
...
 		)
 
+		// Report market as available.
+		availableMarkets[price.MarketId] = true
+
 		pf.writeToBufferedChannel(exchangeId, price, err)
 	}
+
+	// Emit metrics on this exchange's market availability.
+	for marketId, available := range availableMarkets {
```
Comment (teddyding) [inline protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go:313]: For my understanding what is the cadence of this logic? Is it per `runSubTask` run? How often is it?

Comment (clemire) [reply protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go:313]: It's run once every time a price fetcher makes a query.

File: protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go [hunk 4]
```diff
@@ -183,6 +185,27 @@ func (pf *PriceFetcher) RunTaskLoop(requestHandler daemontypes.RequestHandler) {
...
+// Success is tracked by (market, exchange) so that we can track the availability of each market on each exchange.
+func emitMarketAvailabilityMetrics(exchangeId types.ExchangeId, id types.MarketId, available bool) {
+	success := metrics.Success
+	if !available {
+		success = metrics.Error
+	}
+	telemetry.IncrCounterWithLabels(
+		[]string{
+			metrics.PricefeedDaemon,
+			metrics.PriceFetcherQueryForMarket,
```
Comment (teddyding) [inline protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go:198]: Nit: perhaps update the metrics name with a suffix like `_sampled` so it's clear that this is a sampled metric

**Review by teddyding — APPROVED** [review]

File: protocol/lib/metrics/constants.go [hunk 5]
```diff
@@ -343,6 +343,7 @@ const (
 	MarketUpdaterUpdateMarkets              = "market_updater_update_markets"
 	PriceEncoderPriceConversion             = "price_encoder_price_conversion"
 	PriceFetcherQueryExchange               = "price_fetcher_query_exchange"
+	PriceFetcherQueryForMarket              = "price_fetcher_query_for_market"
```
Comment (clemire) [inline protocol/lib/metrics/constants.go:346]:
```suggestion
	PriceFetcherQueryForMarket              = "price_fetcher_query_for_market_sampled"
```

**Pull request merged**